### PR TITLE
#2043 Refactor RPE print roster into something more in line with desired output

### PR DESCRIPTION
**Pull request by adrichey**

### CODE REVIEW

**Comment by stenington** [comment]
@adrichey It seems to me like we can collapse the logic in `printable_scores_controller.rb` down quite a bit, I think it could be reduced to

```
def show
      @event = RegionalPitchEvent.includes(
        judges: :submission_scores,
        team_submissions: :submission_scores
      ).find(params[:id])

      judges = @event.judges

      judge_ids = judges.map { |judge| judge.id }
      @scores = @event.team_submissions
                .flat_map(&:scores)
                .reject { |score| !judge_ids.include?(score.judge_profile_id) }

      @groups = []
      judges.each.with_index do |judge, idx|
        teams = (judge.assigned_teams_for_event(@event) || @event.teams).to_a

        if index = @groups.find_index { |group| group[:teams] == teams }
          @groups[index][:judges].push(judge)
        else
          @groups.push({ judges: [judge], teams: teams })
        end
      end
    end
```

`assigned_teams_for_event` could even return `event.teams` if no assignments are otherwise found, since that seems to be the default.

**Comment by adrichey** [comment]
> @adrichey It seems to me like we can collapse the logic in `printable_scores_controller.rb` down quite a bit, I think it could be reduced to
> 
> ```
> def show
>       @event = RegionalPitchEvent.includes(
>         judges: :submission_scores,
>         team_submissions: :submission_scores
>       ).find(params[:id])
> 
>       judges = @event.judges
> 
>       judge_ids = judges.map { |judge| judge.id }
>       @scores = @event.team_submissions
>                 .flat_map(&:scores)
>                 .reject { |score| !judge_ids.include?(score.judge_profile_id) }
> 
>       @groups = []
>       judges.each.with_index do |judge, idx|
>         teams = (judge.assigned_teams_for_event(@event) || @event.teams).to_a
> 
>         if index = @groups.find_index { |group| group[:teams] == teams }
>           @groups[index][:judges].push(judge)
>         else
>           @groups.push({ judges: [judge], teams: teams })
>         end
>       end
>     end
> ```
> 
> `assigned_teams_for_event` could even return `event.teams` if no assignments are otherwise found, since that seems to be the default.

This is really clean. I believe you are right. I will make the change and run a test and then push that change up.

**Comment by adrichey** [comment]
@stenington This is once again ready for review.

File: app/controllers/regional_ambassador/printable_scores_controller.rb [hunk 1]
```diff
@@ -9,34 +9,26 @@ def show
...
-            j.submission_scores.where(team_submission_id: event_submission_ids)
-          }
+      judge_ids = judges.map { |judge| judge.id }
+      @scores = @event.team_submissions
+                .flat_map(&:scores)
+                .reject { |score| !judge_ids.include?(score.judge_profile_id) }
+
+      @groups = []
+      judges.each.with_index do |judge, idx|
+        if judge.assigned_teams_for_event(@event).any?
```
Comment (stenington) [inline app/controllers/regional_ambassador/printable_scores_controller.rb]: Ah, good catch. I guess this is another argument for `assigned_teams_for_event` defaulting to all the event's teams, but I'm happy with this as is.

**Pull request merged**